### PR TITLE
Implement telemetryservice tdengine

### PR DESCRIPTION
**Pull request by MrLeea-13155bc**
**What this PR does / why we need it**:
Implement telemetryservice tdengine
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [x] unit test
- [x] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: azure-pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -468,18 +468,25 @@ stages:
...
-          docker buildx build -f --platform=linux/amd64 \
+          docker buildx build --platform=linux/amd64 \
           -f $(System.DefaultWorkingDirectory)/examples/telemetryservice/mockserver/Dockerfile.mockserver \
           --build-arg PROJECT_ROOT="$(System.DefaultWorkingDirectory)" $(System.DefaultWorkingDirectory) \
           -t edgehub/mockserver:$(tag) --load
         displayName: build mockServer
       - script: |
           set -e
           docker run -itd --network host --name mosquitto eclipse-mosquitto:2.0.14
+          docker run -itd --network host --name tdengine \
```
Comment (BtXin) [inline azure-pipelines/azure-pipelines.yml:479]: Let's put TDengine's test into a separate stage. If you mix MQTT and TDengine, if the test failed, you need extra time to debug which caused the test fail.

File: examples/httpDeviceShifu/deployment/http-deviceshifu-configmap.yaml [hunk 2]
```diff
@@ -19,16 +19,14 @@ data:
...
           pushSettings:
-            telemetryCollectionService: push-endpoint-2
+           telemetryCollectionService: push-endpoint-1
       device_health2:
         properties:
-          instruction: info
+          instruction: status
           pushSettings:
-            pushToServer: false
+          telemetryCollectionService: push-endpoint-1
```
Comment (BtXin) [inline examples/httpDeviceShifu/deployment/http-deviceshifu-configmap.yaml:32]: Both `device_health1` and `device_health2` are using instruction `status` and push to `push-endpoint-1`?

File: examples/httpDeviceShifu/deployment/http-deviceshifu-telemetryservice.yaml [hunk 3]
```diff
@@ -5,18 +5,12 @@ metadata:
...
-metadata:
-  name: push-endpoint-2
-  namespace: devices
-spec:
-  type: HTTP
-  address: http://192.168.15.48:12345/api1
+    SQLSetting:
+      serverAddress: 192.168.14.163:6041
+      username: root
+      secret: taosdata
```
Comment (BtXin) [inline examples/httpDeviceShifu/deployment/http-deviceshifu-telemetryservice.yaml:13]: Maybe we should encode the secret with k8s secret and get it with ENV_VAR, and here should be the name of the ENV_VAR, and add a comment to state it.

File: examples/telemetryservice/checkoutput.sh [hunk 4]
```diff
@@ -1,14 +1,31 @@
...
     if [[ $output == $default ]]
+    then
+        break
+    elif [[ $i == 100 ]]
+    then
+        exit 1
+    fi
+done
+# init TDEngine Table
+docker exec tdengine taos -f /root/init.sql
```
Comment (BtXin) [inline examples/telemetryservice/checkoutput.sh:21]: Separate TDengine part to a separate file.

**Review by BtXin — COMMENTED** [review]
Discussed offline and added some comments

File: examples/httpDeviceShifu/deployment/http-deviceshifu-telemetryservice.yaml (hunk 3 above)
Comment (MrLeea-13155bc) [reply examples/httpDeviceShifu/deployment/http-deviceshifu-telemetryservice.yaml:13]: handle it after shifuctl implememt

**Comment by BtXin** [comment]
Generally looks good. 
For the password secret part, write a PRD in feishu, create a task then assign to yourself. 
I will give you AP after you finish the above steps.

**Review by BtXin — APPROVED** [review]

**Pull request merged**